tests/components/homekit/test_type_sensors.py

```diff
@@ -68,3 +70,44 @@ def test_humidity(self):
         self.hass.states.set(entity_id, '20', {ATTR_UNIT_OF_MEASUREMENT: "%"})
         self.hass.block_till_done()
         self.assertEqual(acc.char_humidity.value, 20)
+
+    def test_binary(self):
+        """Test if accessory is updated after state change."""
+        entity_id = 'binary_sensor.opening'
+
+        self.hass.states.set(entity_id, STATE_UNKNOWN,
+                             {ATTR_DEVICE_CLASS: "opening"})
+        self.hass.block_till_done()
+
+        acc = BinarySensor(self.hass, entity_id, 'Window Opening', aid=2)
+        acc.run()
+
+        self.assertEqual(acc.get_service(SERV_CONTACT_SENSOR).display_name,
+                         SERV_CONTACT_SENSOR)
+        self.assertEqual(acc.char_detected.display_name,
+                         CHAR_CONTACT_SENSOR_STATE)
```

Comment:
Instead of L85 - L88, why not do something like that:
```py
from homeassistant.components.homekit.type_sensors import BINARY_SENSOR_SERVICE_MAP
from homeassistant.components.homekit.const import (
    DEVICE_CLASS_CO2, SERV_CARBON_DIOXIDE_SENSOR, CHAR_CARBON_DIOXIDE_DETECTED,
    DEVICE_CLASS_GAS, SERV_CARBON_MONOXIDE_SENSOR,
    CHAR_CARBON_MONOXIDE_DETECTED,
    DEVICE_CLASS_MOISTURE, SERV_LEAK_SENSOR, CHAR_LEAK_DETECTED,
    DEVICE_CLASS_MOTION, SERV_MOTION_SENSOR, CHAR_MOTION_DETECTED,
    DEVICE_CLASS_OCCUPANCY, SERV_OCCUPANCY_SENSOR, CHAR_OCCUPANCY_DETECTED,
    DEVICE_CLASS_OPENING, SERV_CONTACT_SENSOR, CHAR_CONTACT_SENSOR_STATE,
    DEVICE_CLASS_SMOKE, SERV_SMOKE_SENSOR, CHAR_SMOKE_DETECTED)

class TestHomekitSensors(unittest.TestCase):

#####
#####

    def test_binary(self):

#####
#####

    def test_binary_device_classes(self):

        entity_id = 'binary_sensor.demo'

        for device_class, (service, char) in BINARY_SENSOR_SERVICE_MAP.items():
            print(device_class)
            self.hass.states.set(entity_id, STATE_OFF,
                                 {ATTR_DEVICE_CLASS: device_class})
            self.hass.block_till_done()

            acc = BinarySensor(self.hass, entity_id, 'Binary Sensor', aid=2)
            self.assertEqual(acc.get_service(service).display_name, service)
            self.assertEqual(acc.char_detected.display_name, char)

```
That way you check if every service and char can be added successfully.

Reply:
done